pom.xml

```diff
@@ -72,7 +72,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <slf4j.version>1.7.21</slf4j.version>
-        <spring.version>4.3.2.RELEASE</spring.version>
+        <spring.version>4.1.0.RELEASE</spring.version>
```

Comment:
But here we may stick with latest spring version because you have implemented backward compatibility, not forward